jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java

```diff
@@ -694,6 +693,58 @@ public Response getInvoicesForAccount(@PathParam("accountId") final UUID account
         return Response.status(Status.OK).entity(result).build();
     }
 
+    @TimedResource
+    @GET
+    @Path("/{accountId:" + UUID_PATTERN + "}/" + INVOICES + "/" + PAGINATION)
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Retrieve account invoices", response = InvoiceJson.class, responseContainer = "List")
+    @ApiResponses(value = {@ApiResponse(code = 400, message = "Invalid account id supplied"),
+                           @ApiResponse(code = 404, message = "Account not found")})
+    public Response getInvoicesForAccountWithPagination(@PathParam("accountId") final UUID accountId,
+                                          @QueryParam(QUERY_INVOICES_FILTER) final String invoicesFilter,
+                                          @QueryParam(QUERY_SEARCH_OFFSET) @DefaultValue("0") final Long offset,
+                                          @QueryParam(QUERY_SEARCH_LIMIT) @DefaultValue("100") final Long limit,
+                                          @QueryParam(QUERY_AUDIT) @DefaultValue("NONE") final AuditMode auditMode,
+                                          @javax.ws.rs.core.Context final HttpServletRequest request) throws AccountApiException {
+
+        final TenantContext tenantContext = context.createTenantContextWithAccountId(accountId, request);
+
+        // Verify the account exists
+        accountUserApi.getAccountById(accountId, tenantContext);
```

Comment:
Interesting -- we typically don't do that though. @sbrossie Any thought?

Reply:
This is already there in the existing `getInvoicesForAccount` method (See this [line](https://github.com/killbill/killbill/blob/c9ca6ca5d5f3310205d6baabee372bb2e007cdea/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java#L689)). I can delete this line from both methods if it is unnecessary.

Reply:
oh, I had no idea! From 2013: https://github.com/killbill/killbill/commit/b15041255b6ffefdc95c6d65027f1f583002e09a#diff-412208bd13addbd00f2f80dde17031143060e334f320967752451977cf8c5e2aR388 😅

Reply:
In don't remember the details, I suspect this was to make sure we were throwing the right `AccountException` and as a result returning the correct http error.

For simplicity, keep it the same - based on time, it would be interesting to try without it, and see what happens with a non existent accountId (if result is the same, remove it).

Reply:
Sure, I'll try deleting that line. If it works, I will create a separate commit.

Reply:
When the endpoint is queried with an invalid `accountId` it displays the following both with and without the account exists check:

```
{"className":"org.killbill.billing.util.callcontext.InternalCallContextFactory$ObjectDoesNotExist","code":null,"message":"Object id=7edbcbfa-da58-492e-a8df-1baadb008485 type=ACCOUNT doesn't exist!","causeClassName":null,"causeMessage":null,"stackTrace":[]}
```

So, looks like the account exists check has no effect. So, I've deleted it as part of [this ](https://github.com/killbill/killbill/pull/1789/commits/f78e2a9278eae56fa45e6d0791c8e922683b9949)commit.